### DISCUSSION TITLE
TsFile查询

### DISCUSSION THREAD

**Discussion by caszhou** [apache/iotdb#4697] [Q&A] [answered]
在不知道path的情况下，有api可以将TsFile转换成CSV（path,time,type,value）么？

**Comment by jixuan1989** [comment, marked as answer]
目前还没有针对TsFile的CSV导出工具，但是可以用iotdb的导出工具来迂回搞定。
1. 把tsfile load到 一个干净的iotdb中， https://iotdb.apache.org/zh/UserGuide/Master/System-Tools/Load-External-Tsfile.html 
2. 使用export-csv工具 使用 "select * from root" 或者 "select * from root align by device"导出数据 https://iotdb.apache.org/zh/UserGuide/Master/System-Tools/CSV-Tool.html

**Reply by caszhou** [reply]
thank you~

**Comment by myysophia** [comment]
> 目前还没有针对TsFile的CSV导出工具，但是可以用iotdb的导出工具来迂回搞定。
> 
> 1. 把tsfile load到 一个干净的iotdb中， https://iotdb.apache.org/zh/UserGuide/Master/System-Tools/Load-External-Tsfile.html
> 2. 使用export-csv工具 使用 "select * from root" 或者 "select * from root align by device"导出数据 https://iotdb.apache.org/zh/UserGuide/Master/System-Tools/CSV-Tool.html

如果不是对齐时间序列，align by device会报错。还有其他什么办法导出大量数据呢
